src/layouts/App/SelectRPCNode.js

```diff
@@ -0,0 +1,84 @@
+import React, { FC, useEffect, useCallback, useState, Dispatch, SetStateAction } from 'react'
+import styled from 'styled-components'
+import { Menu, MenuItem } from './shared'
+import { ENDPOINTS, useConnectionConfig } from '../../context/settings'
+import { notify } from '../../utils'
+import { ArrowDropdown } from '../../components'
+import { SpaceBetweenDiv } from '../../styles'
+
+const WRAPPER = styled(SpaceBetweenDiv)`
+  padding: ${({ theme }) => theme.margin(1.5)};
+  cursor: pointer;
+  ${({ theme }) => theme.smallBorderRadius}
+  background-color: ${({ theme }) => theme.grey5};
+  > span {
+    font-size: 12px;
+    font-weight: bold;
+  }
+`
+
+const ITEM = styled(MenuItem)`
+  width: 140px;
+  > a {
+    display: flex;
+    justify-content: space-between;
+    width: 100%;
+
+    > span {
+      color: ${({ theme }) => theme.text1};
+    }
+  }
+`
+
+const Overlay = ({
+  handleClick
+}: {
+  handleClick: (e: any, endpoint: string, endpointName: string, network: string) => void
+}) => {
+  return (
+    <Menu>
+      {ENDPOINTS.map((point, index) => {
+        return (
+          <ITEM key={index} onClick={(e) => handleClick(e, point.endpoint, point.name, point.network)}>
+            <span>{`${point.name}  (${point.network})`}</span>
+          </ITEM>
+        )
+      })}
+    </Menu>
+  )
+}
+
+export const SelectRPCNode: FC = () => {
+  const context = useConnectionConfig()
```

Comment:
The common pattern we are trying use across the app is the ES6 object de-structuring:
`const { endpoint, endpointName, setEndpoint } = useConnectionConfig()`